codice/ddf
# DDF-1942: Fixed excessive factory method use in the catalog export

#### What does this PR do?
Cleans up the catalog export code according to comments. There were too many code smells. Some factory methods remain and were necessary, but it has considerably improved. 
#### Who is reviewing it?
@stustison @pklinef and @lessarderic 
#### How should this be tested?
Make sure PRBS passes and all bundles come up without issue. Ingest some metacards and invoke the `migration:export` command. Reingest those exported cards after clearing the catalog and ensure nothing was corrupted. 
#### Any background context you want to provide?
Original code written in DDF-1824 needs a little clean-up. 
#### What are the relevant tickets?
DDF-1824 and DDF-1942
#### Screenshots (if appropriate)
#### Checklist:

Updated unit tests accordingly. Coverage does not drop.
